test/functional/wallet_groups.py

```diff
@@ -64,6 +64,52 @@ def run_test(self):
         assert_approx(v[0], 0.2)
         assert_approx(v[1], 1.3, 0.0001)
 
+        # Test 'avoid partial if warranted, even if disabled'
+        self.sync_all()
+        self.nodes[0].generate(1)
+        # Nodes 1-2 now have confirmed UTXOs (letters denote destinations):
+        # Node #1:      Node #2:
+        # - A  1.0      - D0 1.0
+        # - B0 1.0      - D1 0.5
+        # - B1 0.5      - E0 1.0
+        # - C0 1.0      - E1 0.5
+        # - C1 0.5      - F  ~1.3
+        # - D ~0.3
+        assert_approx(self.nodes[1].getbalance(), 4.3, 0.0001)
+        assert_approx(self.nodes[2].getbalance(), 4.3, 0.0001)
+        # Sending 1.4 btc should pick one 1.0 + one more. For node #1,
+        # this could be (A / B0 / C0) + (B1 / C1 / D). We ensure that it is
+        # B0 + B1 or C0 + C1, because this avoids partial spends while not being
+        # detrimental to transaction cost
+        txid3 = self.nodes[1].sendtoaddress(self.nodes[0].getnewaddress(), 1.4)
+        tx3 = self.nodes[1].getrawtransaction(txid3, True)
+        # tx3 should have 2 inputs and 2 outputs
+        assert_equal(2, len(tx3["vin"]))
+        assert_equal(2, len(tx3["vout"]))
+        # the accumulated value should be 1.5, so the outputs should be
+        # ~0.1 and 1.4 and should come from the same destination
+        values = [vout["value"] for vout in tx3["vout"]]
+        values.sort()
+        assert_approx(values[0], 0.1, 0.0001)
+        assert_approx(values[1], 1.4)
```

Comment:
missing `vspan` arg here (iiuc it's the same value as the others, e.g. the default, but since it looks like the intention here is to make the vspans explicit, this one was probably an oversight)

also, can use named args for all of the `assert_approx` calls when more than 1-2 args are passed:
```diff
-        assert_approx(values[0], 0.1, 0.0001)
-        assert_approx(values[1], 1.4)
+        assert_approx(values[0], vexp=0.1, vspan=0.0001)
+        assert_approx(values[1], vexp=1.4, vspan=0.0001)
```